impl/src/main/java/org/ehcache/internal/store/offheap/AbstractOffHeapStore.java

```diff
@@ -843,14 +857,37 @@ public boolean test(Map.Entry<K, OffHeapValueHolder<V>> argument) {
     private void advance() {
       next = null;
       while (next == null && mapIterator.hasNext()) {
-        Map.Entry<K, OffHeapValueHolder<V>> entry = mapIterator.next();
-        final long now = timeSource.getTimeMillis();
-        if (entry.getValue().isExpired(now, TimeUnit.MILLISECONDS)) {
-          containsKey(entry.getKey());  //This is done here to remove the expired entry without any side effects
-          continue;
-        }
+        final Map.Entry<K, OffHeapValueHolder<V>> entry = mapIterator.next();
+        final OffHeapValueHolder<V> value = (OffHeapValueHolder<V>) internalGet(entry.getKey(), false, true);
```

Comment:
This is one of the ugly changes. There is no easy way to tell the iterator to eagerly deserialize and when we have the `OffHeapValueHolder` back, it is out of lock scope so we can't do anything with it. Hence the call to internalGet without updating access stats but forcing deserialization.